build.gradle.kts

```diff
@@ -51,6 +51,7 @@ dependencies {
 tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
    kotlinOptions.jvmTarget = "1.8"
```

Comment:
Btw is there some reason that the jvm target is still 1.8 and not 11? And especially the apiVersion of kotlin? Or I don't what is the policy of libraries usually or the compatibility

Reply:
Java 1.8 as no feature of Java 9+ is used. As a lot of programs are still written / maintained in Java 1.8 this seems a reasonable choice as a library.
But you are right with the kotlin api version. I think it should be changed to 1.4 as the kotlinx.serialization 1.x won't be working with a lower kotlin api.